### PR TITLE
[Sup 1959]Fix Identation issue, add enabling RBAC token step, update task metadata

### PR DESCRIPTION
**Pull request by henrywangpuppet**

### CODE REVIEW

File: tasks/provision_newstyle_compiler.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+declare PT__installdir
+source "$PT__installdir/deploy_pe/files/common.sh"
+PUPPET_BIN=/opt/puppetlabs/bin
+/bin/mkdir -p /root/.puppetlabs
+/bin/curl -X POST -H 'Content-Type: application/json' --cacert $(${PUPPET_BIN}/puppet config print cacert) https://$(${PUPPET_BIN}/puppet config print server):4433/rbac-api/v1/auth/token -d '{"login":"admin","password":"puppetlabs","lifetime":"0"}' | /bin/python -c 'import sys, json; print json.load(sys.stdin)["token"]' > ~/.puppetlabs/token
```
Comment (jarretlavallee) [inline tasks/provision_newstyle_compiler.sh]: This needs some error checking. Please also pass in the password as a parameter since it can be changed.

**Review by jarretlavallee — CHANGES_REQUESTED** [review]
Overall this looks good. I have a couple changes that I would like to see to make it more resilient and flexible.

**Comment by henrywangpuppet** [comment]
Updated the codes already

**Comment by jarretlavallee** [comment]
Looks good. Let's update the reference.md. 

```
gem install puppet-strings
puppet strings generate 
```

**Pull request merged**